src/app/components/tree-view/tree-view.service.ts

```diff
@@ -55,5 +57,23 @@ export class TreeViewService {
         this.nodes.delete(node.nodeIndex);
     }
 
+    public getExpandedNodes(key: string) {
+        const lsKey = "expand-" + key;
+        return this.preferences.get(lsKey, []);
+    }
 
+    public saveToggleState(id: string, key: string) {
+        const lsKey = "expand-" + key;
+        this.preferences.get(lsKey, []).subscribe(el => {
```

Comment:
`this.preferences.get(lsKey, []).first()` so you don't get leftover subscriptions whenever the save is triggered.